simpledotorg/simple-server
# Overdue: Date picker to change number of dashed lines

- MonthsDashedBased is the base number of dashes for a chart
- adding the current date string (todays date Mmm-yyyy) to the function arguments will affect the number of dashed segments depending on how many months between today and the latest month on the chart.

`monthIndexFromChartLabelDateString()` has been moved to a global function as its used in both parts of the reports.js file so I can no longer place it in just one part.

**Story card:** [sc-11007](https://app.shortcut.com/simpledotorg/story/11007/overdue-dashed-lines-showing-incorrectly-on-previous-month-reports)

## Because

Its been highlighted that when selecting an old date on the date picker in the dashboard (eg Jul-2023), the number of dashed line segments show the same even for older dates where the figures are now fixed.
Aug-2023
![Screenshot 2023-08-10 at 11 35 18](https://github.com/simpledotorg/simple-server/assets/9541902/20c2e5d5-6d2c-4af4-ba0a-642f766445b9)
Jul-2023
![Screenshot 2023-08-10 at 11 38 52](https://github.com/simpledotorg/simple-server/assets/9541902/458eeba0-4bd1-4289-8ab1-93c3ea89435f)

## This addresses

It makes the charts confusing to read as the dashboard is stating the there are still changes occurring on older date ranges, which is not the case.

## Test instructions

For Aug-2023
Overdue called - should show 1 dashed segment
Overdue returned: 
- 1st to 15th - 2 dashed lines
- 16th to end of month - 1 dashed line

Now select Jul-2023 from the date picker
Overdue called - should have no dashed lines
Overdue returned: 
- 1st to 15th - 1 dashed line
- 16th to end of month - 0 dashed lines

Now select Jun-2023 from the date picker
Overdue called - should have no dashed lines
Overdue returned - 0 dashed lines

Please check that no errors are thrown with the goal lines activated on all of the date ranges. They share a common function which has change name - `monthIndexFromChartLabelDateString`
Flipper flag: `goal-lines`

You will need to add the following to the config to show goal lines in localhost
in `const enabledRegions`
change the IN line to the following 
`IN: ["organization", "state", "district"],`
